manifests/include_dir.pp

```diff
@@ -0,0 +1,23 @@
+# Add include directories to /etc/sudoers
+#
+# @param include_dir the directory to include in /etc/sudoers
+#
+define sudo::include_dir (
+  Stdlib::Absolutepath $include_dir = '',
+) {
+  include 'sudo'
+
+  file { $include_dir:
```

Comment:
Ideally, you'd set a prefix and be able to optionally tidy files in the directory. Otherwise, anything can update/break the sudoers configuration without warning.

Reply:
We could set this to purge by default, however, we aren't actually managing any directory content in this right now, we're just giving the user the ability to include arbitrary directories via sudo::include_dirs. We'd have to implement the rewrite of the concat fragments into files underneath the include_dir for setting purge on the include_dir to be useful. As it stands right now if a user sets an include_dir to one they have written themselves and don't notice that they need to set tidy_include_dir to false, they could blow away a lot of custom configuration that they didn't mean to. I'm totally fine having the tidy there for right now, but I think it would be dangerous to leave it defaulted to true until we move to using includes over concats.

Reply:
Also, when we discussed adding a prefix I had forgotten that we're actually not leveraging this new define by default, the init.pp sets sudo::include_dirs to a blank array, which means we won't actually be adding any lines into the sudoers file and when we do add those lines, we're not actually managing any of the files beneath the included directory. With that in mind, there's really nothing to prefix here.

Reply:
@michael-riddle That makes sense. As long as the `includedirs` statement is empty by default, then there really isn't anything more to be done.